src/main/webapp/creatives/maps/google_maps_client.js

```diff
@@ -0,0 +1,269 @@
+import {CreativeMessage} from '../constants.js';
+const DEFAULT_MAP_LAT = 37.422004;
+const DEFAULT_MAP_LNG = -122.081402;
+const DEFAULT_ZOOM = 13;
+const DEFAULT_LOCATION_NUM_DISPLAYED = 4;
+const MARKER_SIZE = 25;
+
+/**
+ * This class handles all map related functionality, including
+ * displaying the map, calculating & displaying directions, tracking all
+ * interactions with the map, logging error messages with SimidProtocol,
+ * and handling any errors that result from the Maps API.
+ */
+export default class GoogleMapsClient {
+    /**
+     * A GoogleMapsClient object takes in a userSession object, a search query string, the marker image url,
+     * user coordinates, a SimidProtocol object, and a function that determine's the object's behavior in case
+     * of an error.
+     */
+    constructor(userSession, query, markerUrl,
+         coordinates = new google.maps.LatLng(DEFAULT_MAP_LAT, DEFAULT_MAP_LNG), simidProtocol, onMapsClientComplete) {
+        /**
+         * The LatLng coordinates representing the user's current location.
+         * @private @const {?google.maps.LatLng}
+         */
+        this.currentLocation_ = coordinates;
+        /**
+         * The string representing the search query.
+         * @private @const {?string}
+         */
+        this.searchQuery_ = query;
+        /**
+         * The desired marker image's string URL.
+         * @private @const {?string}
+         */
+        this.markerImage_ = markerUrl;
```

Comment:
Can these ever be null? If not, use `!` instead of `?` ( or for string, number, or boolean types, just leave off the punctuation, since those are assumed to be required unless otherwise specified)

Reply:
Updated the descriptions for all of these